crates/sui-indexer-alt/src/pipeline/sequential/committer.rs

```diff
@@ -319,26 +319,20 @@ pub(super) fn committer<H: Handler + 'static>(
                     batch_rows = 0;
                     attempt = 0;
 
-                    // Keep going if we might get more rows from the processor.
-                    if !rx.is_closed() || !rx.is_empty() {
-                        continue;
-                    }
-
-                    // If there are no more pending checkpoints, stop processing, because there are
-                    // no more coming.
-                    let Some((next, _)) = pending.first_key_value() else {
-                        break;
-                    };
-
-                    // If there is a gap between the next checkpoint and the pending checkpoint,
-                    // stop because there are no more coming.
-                    if next_checkpoint < *next {
-                        break;
-                    }
-
-                    // If checkpoint lag is being enforced, and the pending checkpoint is being
-                    // held back, stop because `commit_hi` will never progress.
-                    if matches!(commit_hi, Some(hi) if hi < *next) {
+                    // If there is a pending checkpoint, no greater than the expected next
+                    // checkpoint, and no less than the upperbound due to checkpoint lag, then the
+                    // pipeline can do more work immediately (without waiting).
+                    //
+                    // Otherwise, if its channels have been closed, we know that it is guaranteed
+                    // not to make any more progress, and we can stop the task.
+                    if pending
+                        .first_key_value()
+                        .is_some_and(|(next, _)| {
+                            *next <= next_checkpoint && commit_hi_inclusive.map_or(true, |hi| *next <= hi)
```

Comment:
`*next <= hi` does not match the comment `no less than the upperbound due to checkpoint lag`

Reply:
The bound is inclusive (see the variable name: `commit_hi_inclusive`)

Reply:
Not about that.
`*next <= hi` would be `no bigger than the upperbound`.

Reply:
Updated the comment to match the variable name.